olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogRequest.h

```diff
@@ -88,7 +91,25 @@ class CatalogRequest final {
     fetch_option_ = fetchoption;
     return *this;
   }
- private:
+
+  /**
+   * @brief Creates readable format for the request.
+   * @param none
+   * @return string representation of the request
+   */
+  std::string CreateKey() const {
```

Comment:
This calls for a base class with abstract CreateKey()

Reply:
Should be marked inline

Reply:
There is no architectural usage base class now, thus abstract interface brings a negligible value.